### PR TITLE
ICU-21450 Add asan check, tsan check, gcc debug build and test, gcc 8…

### PR DESCRIPTION
**Pull request by gnrunge**
… build and

test, and ICU4C release tools buildability check to GitHub Action Continuous Integration.

##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/ICU-21450
- [ ] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: .github/workflows/icu_ci.yml [hunk 1]
```diff
@@ -32,6 +32,31 @@ jobs:
...
+  gcc-debug-build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: ICU4C with gcc
+        run: |
+          mkdir build;
+          cd build;
+          ../icu4c/source/runConfigureICU --enable-debug --disable-release Linux/gcc --prefix=/tmp/icu-prefix --enable-tracing;
```
Comment (echeran) [inline .github/workflows/icu_ci.yml:53]: optional: In the .travis.yml file, and in the job that you added right below this, you made `PREFIX` an env variable and reference it using `${PREFIX}`. You could do that here to for consistency's sake.

File: .github/workflows/icu_ci.yml [hunk 2]
```diff
@@ -58,6 +83,62 @@ jobs:
           CFLAGS: -Wimplicit-fallthrough
           CXXFLAGS: -Wimplicit-fallthrough
 
+  # Out of source build with gcc 8 and c++14; executes icuinfo.
+  gcc-8-stdlib_14:
```
Comment (echeran) [inline .github/workflows/icu_ci.yml:87]: nit: It seems strange to see a YAML key with an underscore. On the other hand, down below in this PR, `INTLTEST_OPTS` is a key that must match the shell env variable to be created.

**Review by echeran — APPROVED** [review]
LGTM

File: .github/workflows/icu_ci.yml (hunk 1 above)
Comment (gnrunge) [reply .github/workflows/icu_ci.yml]: Done.

File: .github/workflows/icu_ci.yml (hunk 2 above)
Comment (gnrunge) [reply .github/workflows/icu_ci.yml]: Changed to 'stdlib14'.

**Pull request merged**